test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -4202,6 +4208,361 @@ defmodule Trento.SapSystemTest do
     end
   end
 
+  describe "delta deregistration" do
+    test "newly registered instances should be marked as present" do
```

Comment:
Do we need this 1st test?
We have plenty of tests where we have a regular registration. We could add the `absent_at` pattern matching in any of them.

Reply:
If you still want to keep it, i think we should still split the database and application parts.
But the test must be fixed in any case.
The `database_instance_registered` must not go in the initial events, as this is the event which triggers the change, so I would suggest to use the command in the 2nd param of `assert_events_and_state`